src/webgpu/util/math.ts

```diff
@@ -288,16 +295,57 @@ export function linearRange(a: number, b: number, num_steps: number): Array<numb
  * thus biasing towards 0, but remaining on the [0, 1] range.
  * This biased range is then scaled to the desired range using lerp.
  * Different curves could be generated by changing c, where greater values of c will bias more towards 0.
- * */
+ */
 export function biasedRange(a: number, b: number, num_steps: number): Array<number> {
   const c = 2;
-  assert(num_steps > 0, '|num_steps| must be greater than 0');
+  if (num_steps <= 0) {
+    return Array<number>();
+  }
+
+  // Avoid division by 0
+  if (num_steps === 1) {
+    return [a];
+  }
 
   return Array.from(Array(num_steps).keys()).map(i =>
     lerp(a, b, Math.pow(lerp(0, 1, i / (num_steps - 1)), c))
   );
 }
 
+/**
+ * @returns an ascending sorted array of numbers spread over the entire range of 32-bit floats
+ *
+ * Numbers are divided into 4 regions: negative normals, negative subnormals, positive subnormals & positive normals.
+ * Zero is included. The normal number regions are biased towards zero, and the subnormal regions are linearly spread.
+ *
+ * @param neg_count number of numbers to generate in the negative normals region
+ * @param neg_sub_count number of numbers to generate in the negative subnormals region
+ * @param pos_sub_count number of numbers to generate in the positive subnormals region
+ * @param pos_count number of numbers to generate in the positive normals region
+ */
+export function fullF32Range(
+  neg_count: number = 50,
```

Comment:
What do you think about using a structure parameter instead of 4 values?
With a structure:
* You're forced to name the arguments, helping with readability of the code. I would have to look up the function to understand what each of the values in `fullF32Range(250, 20, 20, 250)` means.
* You can default any of the 4 values by omitting them. With parameter defaults like this, you can only omit `pos_sub_count` if you provide `neg_count` and `neg_sub_count`, etc.